### PR TITLE
IGAPP-1283: Fix flickering header text

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by steffenkleinle** [comment]
Things done:
- Fix flickering header by saving the previous route to the state and make the routes responsible for setting the correct header
- Open news details in a separate route instead of just changing the params for more consistent behavior and logic (also includes not showing the news header in the detail view)
- Removing the SettingsHeader since it was nearly completely redundant

**Review by sarahsporck — APPROVED** [review]
Tested on Android Emulator.
Works nice! Good job. 
To me it feels counterintuitive to display a route title in the impressum but not in the settings. I think it is better to stay consistent here. Though I have no preference, which one is better :).

**Comment by steffenkleinle** [comment]
> Tested on Android Emulator. Works nice! Good job. To me it feels counterintuitive to display a route title in the impressum but not in the settings. I think it is better to stay consistent here. Though I have no preference, which one is better :).

Sorry for rebasing! I made it consistent in all routes now to show the header title (also for settings, imprint and open source licenses).

File: native/src/components/Header.tsx [hunk 1]
```diff
@@ -79,13 +76,15 @@ const Header = ({
...
-  city,
+  showOverflowItems = true,
   languages,
   isHome,
 }: HeaderProps): ReactElement | null => {
   const { t } = useTranslation('layout')
   const showSnackbar = useSnackbar()
   const deviceWidth = useWindowDimensions().width
+  // Save previous route to state to prevent it from changing during navigating which would lead to flickering of the title
+  const [previousRoute] = useState(navigation.getState().routes[navigation.getState().routes.length - 2])
```
Comment (LeandraH) [inline native/src/components/Header.tsx:87]: Nice!

File: release-notes/unreleased/IGAPP-1283-header-title-flickering.yml
Comment (LeandraH) [inline release-notes/unreleased/IGAPP-1283-header-title-flickering.yml:1]: Thanks 😅

**Review by LeandraH — APPROVED** [review]
Nice, tested on emulated iOS

File: release-notes/unreleased/IGAPP-1283-header-title-flickering.yml
Comment (LeandraH) [reply release-notes/unreleased/IGAPP-1283-header-title-flickering.yml:1]: Oh wait, it should probably have the correct title, though

Comment (steffenkleinle) [reply release-notes/unreleased/IGAPP-1283-header-title-flickering.yml:1]: deleted, just copy and pasted twice

**Pull request merged**